planner/core/rule_join_reorder.go

```diff
@@ -160,14 +210,8 @@ func (s *baseSingleGroupJoinOrderSolver) makeBushyJoin(cartesianJoinGroup []Logi
 				resultJoinGroup = append(resultJoinGroup, cartesianJoinGroup[i])
 				break
 			}
-			newJoin := s.newCartesianJoin(cartesianJoinGroup[i], cartesianJoinGroup[i+1])
-			for i := len(s.otherConds) - 1; i >= 0; i-- {
-				cols := expression.ExtractColumns(s.otherConds[i])
-				if newJoin.schema.ColumnsIndices(cols) != nil {
-					newJoin.OtherConditions = append(newJoin.OtherConditions, s.otherConds[i])
-					s.otherConds = append(s.otherConds[:i], s.otherConds[i+1:]...)
-				}
-			}
```

Comment:
This can be changed or not, what is your opinion? @qw4990 @winoros

Reply:
Actually, I'm not sure, but to make it easier to maintain, I think all join-creating operations should be handled by a standard and dedicated function, so I prefer to use `MakeJoin` here.